tests/subsys/rtio/rtio_api/src/test_rtio_api.c

```diff
@@ -458,14 +424,49 @@ ZTEST(rtio_api, test_rtio_transaction)
 
 	TC_PRINT("rtio transaction simple\n");
 	for (int i = 0; i < TEST_REPEATS; i++) {
-		test_rtio_transaction_(&r_transaction_simp);
+		test_rtio_transaction_(&r_transaction);
 	}
-	TC_PRINT("rtio transaction concurrent\n");
-	for (int i = 0; i < TEST_REPEATS; i++) {
-		test_rtio_transaction_(&r_transaction_con);
+}
+
+#define THROUGHPUT_ITERS 100000
+RTIO_DEFINE(r_throughput, 4, 4);
+
+void _test_rtio_throughput(struct rtio *r)
+{
+	timing_t start_time, end_time;
+	struct rtio_cqe *cqe;
+	struct rtio_sqe *sqe;
+
+	timing_init();
+	timing_start();
+
+	start_time = timing_counter_get();
+
+	for (uint32_t i = 0; i < THROUGHPUT_ITERS; i++) {
+		sqe = rtio_sqe_acquire(r);
+		rtio_sqe_prep_nop(sqe, NULL, NULL);
+		rtio_submit(r, 0);
+		cqe = rtio_cqe_consume(r);
+		rtio_cqe_release(r, cqe);
 	}
+
+	end_time = timing_counter_get();
+
+	uint64_t cycles = timing_cycles_get(&start_time, &end_time);
+	uint64_t ns = timing_cycles_to_ns(cycles);
+
+	zassert_true(ns < UINT64_C(1000000000), "Should take less than a second");
```

Comment:
Looks like this is failing

Reply:
Yeah right, on renode, which isn't easily ignored by the test case. Probably should remove the assert here? Up to you, I added them at your request :-)

Reply:
Feels odd to have a test with no asserts, I'll let you make the call on that. I do want to look at benchmarking tests but that's not slotted for me until 2024 :)